src/components/Banner/BannerCollapseButton.tsx

```diff
@@ -0,0 +1,22 @@
+import type { FC, MouseEventHandler } from 'react';
+import type { ButtonProps } from '../Button';
+import { Button } from '../Button';
+
+export type BannerCollapseButtonProps = ButtonProps;
+
+export const BannerCollapseButton: FC<BannerCollapseButtonProps> = ({ children, ...props }) => {
+  const onClick: MouseEventHandler<HTMLButtonElement> = (event) => {
+    const collapseButton = event.target as HTMLButtonElement;
+    const parentBanner = collapseButton.closest('[data-testid="flowbite-banner"]');
```

Comment:
I'm not sure if using the data-testid here for implementation logic is advisable.

Maybe use the `role=banner` here to identify the parent

Reply:
Agree now that we have that, will change that as well now.